### PR TITLE
 New Tool: OMERO region of interest (ROI) upload using ezomero

### PR DESCRIPTION
**Pull request by rmassei**
New tool of the OMERO suite to upload region of interests (ROI) in a user-defined OMERO.server using [ezomero](https://github.com/TheJacksonLaboratory/ezomero)

- The tool is taking as input a .tsv file containing information to create a ROI (i.e. Polygon, Ellipse, Points) in OMERO. The ROI will be then associated to a specific image by imputing its ID.

image
image

- Output is a .txt report with the ROI ID uploaded into OMERO.

**EDIT:** 

This PR fix also the *$__user__.extra_preferences.get* in all xml files:

```
        --user $__user__.extra_preferences.get('omero_account|username', $test_username)
        --psw $__user__.extra_preferences.get('omero_account|password', $test_password)
```

### CODE REVIEW

File: tools/omero/omero_roi_upload.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
+import argparse
+import re
+
+from ezomero import post_roi, connect
+from ezomero.rois import Ellipse, Label, Line, Point, Polygon, Polyline, Rectangle
+import pandas as pd
```
Comment (bernt-matthias) [inline tools/omero/omero_roi_upload.py:6]:
```suggestion
import pandas as pd
from ezomero import connect,post_roi
from ezomero.rois import Ellipse, Label, Line, Point, Polygon, Polyline, Rectangle
```

**Comment by rmassei** [comment]
Finally, I changed the tests output taking into consideration the OMERO dummy dataset (we start now we already some existing IDs). 
I still not implement the new tool testing based on the regex, I will do in the next PR :)

File: tools/omero/omero_import.xml [hunk 2]
```diff
@@ -25,8 +25,8 @@
         omero import folder -T Dataset:name:$dataset_name
             -s $omero_host
             -p $omero_port
-            -u $__user__.extra_preferences.get('omero|username', $test_username)
```
Comment (bernt-matthias) [inline tools/omero/omero_import.xml:28]: We need to bump the version suffix of the tool.

File: tools/omero/omero_metadata_import.xml [hunk 3]
```diff
@@ -15,8 +15,8 @@
     </requirements>
     <command detect_errors="exit_code"><![CDATA[
         python $__tool_directory__/omero_metadata_upload.py
-        --user $__user__.extra_preferences.get('omero|username', $test_username)
```
Comment (bernt-matthias) [inline tools/omero/omero_metadata_import.xml:18]: Same here.

File: tools/omero/omero_roi_import.xml [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        <token name="@TOOL_VERSION@">5.18.0</token>
+        <token name="@VERSION_SUFFIX@">0</token>
+    </macros>
+    <xrefs>
+        <xref type="bio.tools">omero</xref>
+    </xrefs>
+    <requirements>
+        <requirement type="package" version="3.0.1">ezomero</requirement>
+        <requirement type="package" version="2.2.2">pandas</requirement>
+        <requirement type="package" version="21.0.2">openjdk</requirement>
```
Comment (bernt-matthias) [inline tools/omero/omero_roi_import.xml:14]: Still confused why java is needed here. Could you add a comment?

File: tools/omero/omero_roi_import.xml [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+        python $__tool_directory__/omero_roi_upload.py
+        --input_file $input
+        --image_id $id
+        --user $__user__.extra_preferences.get('omero_account|username', $test_username)
+        --psw $__user__.extra_preferences.get('omero_account|password', $test_password)
+        --host $omero_host
+        --port $omero_port
+        --log_file $log
```
Comment (bernt-matthias) [inline tools/omero/omero_roi_import.xml:24]: Always single-quote everything that could contain spaces. 

```suggestion
        python '$__tool_directory__/omero_roi_upload.py'
        --input_file '$input'
        --image_id $id
        --user '$__user__.extra_preferences.get("omero_account|username", $test_username)'
        --psw '$__user__.extra_preferences.get("omero_account|password", $test_password)'
        --host '$omero_host'
        --port $omero_port
        --log_file '$log'
```

File: tools/omero/omero_roi_import.xml [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+    ]]></command>
+    <inputs>
+        <param argument="input" type="data" format="tabular" optional="false" label="Tab File with ROIs" help="Select ROIs Tabular file"/>
+        <param argument="id" type="integer" optional="false" label="Image ID where annotate the ROIs"/>
+        <param argument="omero_host" type="text" optional="false" label="OMERO host URL"/>
```
Comment (bernt-matthias) [inline tools/omero/omero_roi_import.xml:29]: An validator would be good to have here (and in the other tools), i.e. such that users can only enter valid host names. Could also be less strict, e.g. space free string consisting of `[\w-/:]` or something like this. Plus a validator checking that the string is not empty. 

Ideally implemented as xml macro.

File: tools/omero/omero_roi_import.xml (hunk 4 above)
Comment (rmassei) [reply tools/omero/omero_roi_import.xml:15]: Added also to the other xml file

File: tools/omero/omero_import.xml (hunk 2 above)
Comment (rmassei) [reply tools/omero/omero_import.xml:28]: Done

**Review by bernt-matthias — APPROVED** [review]

**Comment by bernt-matthias** [comment]
Can you also open a PR against https://github.com/Helmholtz-UFZ/ufz-galaxy-tools

**Comment by rmassei** [comment]
> Can you also open a PR against https://github.com/Helmholtz-UFZ/ufz-galaxy-tools

https://github.com/Helmholtz-UFZ/ufz-galaxy-tools/pull/20

**Pull request merged**